pkg/api/server.go

```diff
@@ -56,6 +56,11 @@ type ApiServerConfig struct {
 const HEALTH_CHECK_ROUTE = "/health"
 
 func NewApiServer(config ApiServerConfig) *ApiServer {
+	_, err := net.Listen("tcp", fmt.Sprintf(":%d", config.ApiPort))
```

Comment:
Just noticed this now, sorry. Please change this to `net.Dial` (and in the other places)

Listen spawns a listener that should be handled.

Reply:
Done